tests/python/codegen/test_inject_ptx_ldg32.py

```diff
@@ -20,7 +20,8 @@
 import tvm.testing
 
 
-@T.prim_func
+# A_local is undefined
+@T.prim_func(check_well_formed=False)
```

Comment:
Any suggestion for avoiding this undefined var? Would `decl_buffer` work?

Reply:
Good to know, I'll do that

Reply:
Hm, possible parsing bug? I get this error:
```python
ValueError: Block frame or PrimFunc frame not find. Please ensure 'T.alloc_buffer' is called under T.block() or T.prim_func()
 --> /home/slyubomirsky/code/tvm/tests/python/codegen/test_inject_ptx_ldg32.py:31:15
    |  
 31 |      A_local = T.alloc_buffer((32), "float32", scope="local")
    |                ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
```
It is being done inside a PrimFunc, so I don't know why we would get that error. I'll add a block and see if the example will work regardless.

Reply:
Whoops, you were right initially, and it should have been a `T.decl_buffer`.  I got it confused between the `T.decl_buffer` construct in TVMScript, and the `DeclBuffer` node in C++.  If `T.decl_buffer` has the default value of `data=None`, then it will expand to a `DeclBuffer` and the corresponding `Allocate` node.

The `T.alloc_buffer` is only valid within a block, and fills the `BlockNode::alloc_buffers` field.